### DISCUSSION TITLE
IntelliJ: Module file location

### DISCUSSION THREAD

**Discussion by tcf909** [invertase/melos#337] [Q&A]
Any way to set the module files for IntelliJ generated files to be within the .idea folder? We typically move the *.iml files to the .idea folder and remap the references appropriately.

**Comment by blaugold** [comment]
Unfortunately, there isn't right now.

**Reply by ryanhanks** [reply]
I can see the value in consolidating all the intellij-related generated files into the .idea forlder, assuming that's technically possible. For me personally, I would prefer to have all the generated .iml consolidated underneath a single directory that I rarely if ever look at, and feel comfortable deleting / recreating on demand. I generally see the files as noise.

I wonder what the typical user is like in this regard though. Do we have a hunch on whether people keep .idea or .iml assets under revision control? Based on the popularity of gitignore entries for these types of files, we might assume that people aren't tracking these on the average, but from tcf909's question it's not obvious to me as to whether they are or not.

What constraints are we working within in terms of the directories into which these files are generated? Is this situation similar at all to the existence of the "melos_" prefix being a mechanism to avoid clobber, and we need to avoid stepping on toes when deciding where to put these files? Or might we have a little more freedom here, and potentially have the possibility to migrate towards a default behavior of generating all .iml files into the .idea folder?

Regardless of what needs to be maintained as default behavior, one possible solution I see that could satisfy this inquiry is generating a shadow directory tree in .idea for all the .iml files for the workspace (i.e., `.idea/melos` or something like that). So for example an iml file for a package in 'packages/project1' could end up in '.idea/melos/packages/project1/project.iml'.

To be clear, I don't know this to be technically possible, just going off what was said by tcf909 in their question.
